test/indexer/__init__.py

```diff
@@ -66,7 +70,17 @@ def _create_writer(self, catalog: Optional[CatalogName]) -> IndexWriter:
         return writer
 
 
-class CannedBundleTestCase(AzulUnitTestCase):
+BUNDLE = TypeVar('BUNDLE', bound=Bundle)
+
+
+class CannedBundleTestCase(AzulUnitTestCase, Generic[BUNDLE]):
+
+    @classmethod
+    @abstractmethod
+    def _bundle_cls(cls) -> Type[BUNDLE]:
+        raise NotImplementedError
+
+    data_prefix = Path(__file__).resolve().parent / 'data'
```

Comment:
> The rename of file 826dea02-e274-affe-aabc-eb3db63ad068.tdr.json confused me for bit

The intent was for all canned bundles saved as `*.result.tdr.json` to be renamed to `*.tdr.json` for consistency with the new `*.dss.json` files. Unfortunately, it looks like I accidentally changed the contents of the existing HCA canned while trying to migrate them (I had automated the process), causing the TDR HCA bundles to not register as renamings in the diff. So it's actually very helpful and important that you pointed it out, because it means I need to fix those unintended changes which will hopefully shrink the diff by a huge margin.

Reply:
>  It looks like it might have been previously misnamed?

The `.result.tdr.json` naming convention was poorly thought out and has been a thorn in my side for a long time (I was the one who came up with it). So it wasn't misnamed per se, but our previous convention had some shortcomings that I'm finally being forced to address now that we also need to integrate with AnVIL bundles into the new convention.

Reply:
> Consider condensing into a single commit the changes in 'Fix incorrect types in Config.Catalog doctest' and 'Narrow return type', one commit about general type hinting and misc. fixes or improvements

I disagree. I think the history is easier to read (and the PR is therefore easier to review) if each commit has a clear and unique purpose, even if the change expressing that purpose is very small.

Reply:
>and the file rename could also be included in this commit. I might be misunderstanding something, so don't hesitate in letting me know.

No, the renaming definitely belongs in the commit that changes the other canned bundles, otherwise the tests will use the wrong file name and fail to load the file.